libraries/AP_Notify/Buzzer.cpp

```diff
@@ -99,40 +142,46 @@ void Buzzer::update_pattern_to_play()
         if (_flags.ekf_bad) {
             // ekf bad warning buzz
             play_pattern(EKF_BAD);
+            return;
         }
-        return;
     }
 
     // if vehicle lost was enabled, starting beep
     if (AP_Notify::flags.vehicle_lost) {
         play_pattern(DOUBLE_BUZZ);
         return;
     }
-
-    // if battery failsafe constantly single buzz
-    if (AP_Notify::flags.failsafe_battery) {
-        play_pattern(SINGLE_BUZZ);
-        return;
-    }
 }
 
 
 void Buzzer::update_playing_pattern()
 {
-    if (_pattern == 0UL) {
+    // if no '1's to play in pattern and buzzer not on then just return
+    if (_pattern == 0UL && !_flags.on) {
         return;
     }
 
-    const uint32_t now = AP_HAL::millis();
-    const uint32_t delta = now - _pattern_start_time;
-    if (delta >= 3200) {
-        // finished playing pattern
-        on(false);
-        _pattern = 0UL;
+    // reduce 50hz call down to 10hz
```

Comment:
We've already discussed reverting these changes, and using find-first-bit methods to shorten the time requires to play things.  I'm happy to do that last bit if you'd prefer not to.

Reply:
I've coded a way to do it using the timing as requested; just need to do some testing.